### ISSUE TITLE
is useIntl() memoized?

### ISSUE THREAD

**Issue by c3s4** [formatjs/formatjs#1631]
I'm using `useIntl` to get `intl` instance to format the messages. I need to use the `intl`instance inside an `useEffect` so it has to be in the dependency array. This causes an infinite render loop.

I have no idea how to avoid this. Yes I can not list as dependency but is a bad practice and the linter complains about this. Maybe a memoized version could be the right way. I can't imagine why the instance have to change during the app life, but I didn't dig into source code.

**Comment by longlho** [comment]
Do you have a codesandbox to repro?

**Comment by c3s4** [comment]
I was trying to create a codesandbox but it was causing a loop as expected, then I tried to create something usable and in the end I figured out the loop does not happen again. I investigated a little more and I figured out that this issue is related to my messages object. For some strange reason (maybe is related to the react router, but I have to investigate more) at each render my object is recreated and this causes a new `intl` instance. 
Maybe is only my fault, sorry. I think is correct that if the message object changes we have another `intl` instance.
Sorry again
